### DISCUSSION TITLE
Are there a way to set a extrude length on unload filament macro?

### DISCUSSION THREAD

**Discussion by dhnsilva** [jschuh/klipper-macros#118] [Q&A] [answered]
I am encountering problems in unload filament. In some cases on unload filament, the filament gets stuck because the end has a thickest part. This is easely solved extruding a little more material when the filament is unloaded. 
I cant find the portion of the code that treats this subject.

Someone can help me?

**Comment by jschuh** [comment, marked as answer]
This is the spot where the `_LOAD_UNLOAD` macro extrudes a small amount of filament before performing the full unload: https://github.com/jschuh/klipper-macros/blob/a7c928a334c5c63cbb07654c8dd5cc49f33c9b35/filament.cfg#L57

There are other places in the code like this, with small extrusions or retractions (e.g. [to reduce ooze when parking](https://github.com/jschuh/klipper-macros/blob/a7c928a334c5c63cbb07654c8dd5cc49f33c9b35/start_end.cfg#L385)). I've considered switching it all over to a single value controlled by a configuration parameter, because on my own machines I have some volcano nozzles that could use a bit more retraction to avoid oozing. I'm just not sure if a single value would work for all cases, or if the added configuration complexity is warranted.
